lib/utils/listener.go

```diff
@@ -37,3 +39,32 @@ func GetListenerFile(listener net.Listener) (*os.File, error) {
 	}
 	return nil, trace.BadParameter("unsupported listener: %T", listener)
 }
+
+// SingleUseListener wraps a single [net.Conn] and returns it from Accept()
```

Comment:
moved from lib/srv/db/spanner/listener.go

Reply:
Is there a more appropriate package than `lib/utils` for this?

`lib/utils` currently depends on a number of misc packages, including kingpin, 20+ AWS SDK packages, a bunch of protobuf and gRPC packages, a handful of kubernetes packages, OpenTelemetry, go-saml, etc..

Further contributing to this bloat means that anyone who wants this new tool gets all of the above in their binary as well.

Reply:
I couldn't find an existing package for this. 

How about a new `listenerutils` package. We do have a few other useful listener wrappers around.

Alternatively, we can do `netutils` (but i feel this is too generic).

Reply:
👍  For not adding more to `lib/utils`

Reply:
Will move to `lib/utils/listener`